### PR TITLE
chore(npm): updated webpack version and config cleanup

### PR DESCRIPTION
**Pull request by TheLarkInn**
* webpack .25 has a significant breaking changes that requires reworking of configuration

### CODE REVIEW

File: packages/angular-cli/models/webpack-build-common.ts [hunk 1]
```diff
@@ -46,16 +45,15 @@ export function getWebpackCommonConfig(
       filename: '[name].bundle.js'
     },
     module: {
-      preLoaders: [
+      loaders: [
         {
+          enforce: "left",
```
Comment (sokra) [inline packages/angular-cli/models/webpack-build-common.ts]: preLoader -> enforce: "right"

Comment (TheLarkInn) [inline packages/angular-cli/models/webpack-build-common.ts]: Thanks I updated this

File: package.json [hunk 2]
```diff
@@ -109,7 +109,7 @@
     "typedoc": "^0.4.2",
     "typescript": "2.0.2",
     "url-loader": "^0.5.7",
-    "webpack": "2.1.0-beta.22",
+    "webpack": "^2.1.0-beta.23",
```
Comment (sokra) [inline package.json]: better fix the version

File: package.json [hunk 3]
```diff
@@ -157,6 +157,7 @@
     "tree-kill": "^1.0.0",
     "ts-node": "^1.3.0",
     "tslint": "^3.11.0",
-    "walk-sync": "^0.2.6"
+    "walk-sync": "^0.2.6",
+    "webpack": "^2.1.0-beta.23"
```
Comment (sokra) [inline package.json]: why dependency and devDependency

Comment (TheLarkInn) [inline package.json]: The angular-cli creates a new project, and the repo itself is a dev-dep of the generated project. So in the scope of the `angular-cli`, webpack is a dependency, not a dev dependency.

Comment (TheLarkInn) [inline package.json]: A new project may look like this. 

screen shot 2016-09-20 at 10 36 30 am

File: package.json (hunk 2 above)
Comment (TheLarkInn) [inline package.json]: Was working on that as we speak lol!

**Comment by leon** [comment]
They renamed left/right to pre/post

https://github.com/webpack/webpack/releases/tag/v2.1.0-beta.24

**Comment by TheLarkInn** [comment]
@leon thanks, I'm aware I'm on the webpack core team!!! 😅 😅  Just hadn't had the time to get back to the PR with some <3<3

**Comment by TheLarkInn** [comment]
@hansl after careful inspection, I have been able on a test project to get this to run successfully, but have no idea what is causing the e2e test failures. One thing I have noticed is that whenever I run the e2e test runner locally I find that the builds are still running against webpack@2.1.0-beta.22 when they should be running against .25. See below. 
screen shot 2016-09-27 at 9 27 13 am

**Comment by TheLarkInn** [comment]
Well hold that thought @filipesilva just mentioned that there are two package.json 🔥  that I have to update this in. I'm going to try this out locally first.

**Pull request merged**